### PR TITLE
[Torch Dialect] Add support for aten::glu

### PR DESCRIPTION
**Pull request by Groverkss**
This patch adds support for aten::glu

The implementation is based on: https://github.com/pytorch/pytorch/blob/73bf32cb5709b85ab7df4f986d259612073eb0fc/aten/src/ATen/native/GatedLinearUnit.cpp#L23

### CODE REVIEW

**Comment by Groverkss** [comment]
I was not sure where to put this operation in test_suite, so I put it in basic.py for now. Suggestions on where I should put this instead would be helpful.

I did not add this operation to the abstract interp lib because I was not sure if that is required for decompositions. If it is needed, please let me know, I will add that too.

File: python/torch_mlir_e2e_test/test_suite/basic.py
Comment (makslevental) [inline python/torch_mlir_e2e_test/test_suite/basic.py:3269]: as discussed, can you please add a test that exercises this lowering through the public [torch api](https://pytorch.org/docs/stable/generated/torch.nn.GLU.html)

**Review by makslevental — CHANGES_REQUESTED** [review]
LGTM (modulo that one test) but should have another set of eyes @vivekkhandelwal1 can you PTAL

**Comment by makslevental** [comment]
@Groverkss @vivekkhandelwal1 also I believe that indeed it is the case that you need to update the shape library for decomps? https://github.com/llvm/torch-mlir/pull/1804/files

**Comment by vivekkhandelwal1** [comment]
@Groverkss Please add the shape and dtype info for the op as specified here: https://github.com/llvm/torch-mlir/blob/main/docs/abstract_interp_lib.md

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp
Comment (vivekkhandelwal1) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:3979]: The op decomposed also needs to be added here: https://github.com/llvm/torch-mlir/blob/main/lib/Dialect/Torch/Transforms/LowerToBackendContract.cpp#L443 like other ops.

File: lib/Conversion/TorchToArith/TorchToArith.cpp
Comment (ramiro050) [inline lib/Conversion/TorchToArith/TorchToArith.cpp:144]: Drive by comment: this op also needs an e2e test. See: https://github.com/llvm/torch-mlir/blob/main/docs/architecture.md#dos-and-donts-for-unit-vs-end-to-end-testing

> DO use an [end-to-end test](https://github.com/llvm/torch-mlir/blob/main/docs/adding_an_e2e_test.md) if you are implementing a new op or extending the support for an existing op.

File: python/torch_mlir_e2e_test/test_suite/scalar.py
Comment (ramiro050) [inline python/torch_mlir_e2e_test/test_suite/scalar.py:448]: nit: replace `float` with `int`

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:3816]: nit: I believe all these ops have the same result type, so I would create a variable `Value resultType = op.getResult().getType()`, and use that for all of them to make things clear

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py
Comment (ramiro050) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py:583]: This needs to be upstreamed first. See: https://github.com/llvm/torch-mlir/blob/main/docs/adding_abstract_interpretation_functions.md#step-by-step-guide

> New shape functions should be added upstream following the example of https://github.com/pytorch/pytorch/pull/76889, though it can be useful to iterate locally in abstract_interp_lib_gen.py first.

**Review by ramiro050 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
